pkgs/development/python-modules/python-crontab/default.nix

```
@@ -0,0 +1,27 @@
+{ lib, buildPythonPackage, fetchPypi, python-dateutil, pytest }:
+
+buildPythonPackage rec {
+  pname = "python-crontab";
+  version = "2.5.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "4bbe7e720753a132ca4ca9d4094915f40e9d9dc8a807a4564007651018ce8c31";
+  };
+
+  checkInputs = [ pytest ];
+  checkPhase = "pytest --deselect=tests/test_compatibility.py::CompatTestCase::test_07_non_posix_shell";
```

Using the `pytestCheckHook` is preferrential. https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook

```suggestion
  checkInputs = [ pytestCheckHook ];
  
  pytestFlagsArray = [ "--deselect=tests/test_compatibility.py::CompatTestCase::test_07_non_posix_shell" ];
```

---

or `disabledTests = [ "test_07_non_posix_shell"]`

---

Yep, didn't want to assume the tests name was unique, since they were so very specific with the deselect.

---

that's fair, but i would say the average test suite isn't likely to have that happen

---

Thanks @mweinelt and @jonringer for comments. Have now updated to use `disabledTests`. At the moment it is only one test with this name. If the package get another test with same name at some point. We will probably also disable it, since we have "non_posix" in the name.